### ISSUE TITLE
Fails to parse sample YAML v2 document

### ISSUE THREAD

**Issue by pwoolcoc** [wafflespeanut/paperclip#196]
The sample YAML document on this page: https://swagger.io/docs/specification/2-0/basic-structure/ cannot be deserialized by paperclip, it errors out here:

```
---- test_basic_structure stdout ----
thread 'test_basic_structure' panicked at 'Couldn't deser doc: Yaml(Message("invalid type: integer `200`, expected a string", Some(Pos { marker: Marker { index: 183, line: 14, col: 7 }, path: "paths./users" })))', src/lib.rs:41:45
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Here is the code I used to generate this error:

```
use paperclip::v2::{self, ResolvableApi, DefaultSchema, models::Version};

#[test]
fn test_basic_structure() {
    use std::io::{self, Cursor};

    let doc = r#"swagger: "2.0"
info:
  title: Sample API
  description: API description in Markdown.
  version: 1.0.0

host: api.example.com
basePath: /v1
schemes:
  - https

paths:
  /users:
    get:
      summary: Some text
      description: Some more text
      produces:
        - application/json
      responses:
        200:
          description: OK"#.to_string();

    let bytes = doc.as_bytes().to_vec();
    let mut bytes = Cursor::new(bytes);

    let api: ResolvableApi<DefaultSchema> = v2::from_reader(&mut bytes).expect("Couldn't deser doc");
    assert_eq!(api.swagger, Version::V2);
}
```

**Comment by wafflespeanut** [comment]
As the error says, you should have `"200"` (string) instead of `200` (integer) in responses schema.

**Comment by pwoolcoc** [comment]
ok, but this sample is not from me, but from the official OpenAPIv2 spec...not accepting `200` means paperclip is not spec-compliant.

**Comment by wafflespeanut** [comment]
The official spec seems to describe the status code as a string (example [here](https://github.com/OAI/OpenAPI-Specification/blob/c5e9a8a703f40da5d5a1a63dc1ff7530263eca4f/versions/2.0.md#responses-object-example)). The JSON schema for v2 spec also says that the [responses schema is an object](https://github.com/OAI/OpenAPI-Specification/blob/c5e9a8a703f40da5d5a1a63dc1ff7530263eca4f/schemas/v2.0/schema.json#L202), which only have string keys.

**Comment by pwoolcoc** [comment]
exactly, so the `200` should be parsed as a string instead of integer, but it's not, that's where the issue is.

**Comment by wafflespeanut** [comment]
Um, no. The `200` should be written as a string in the spec (it will be parsed as an integer, but it should still be `"200"` in the spec).

**Comment by mxndtaylor** [comment]
So I've looked into this, and there seems to be an issue with `serde`. Specifically, a `struct` that has a `#[serde(flatten)]` property of `BTreeMap<_, BTreeMap<String, _>>` doesn't automatically cast numbers to `String`s, while a `struct` with the same property without a `#[serde(flatten)]` attribute does.

Example:
https://gist.github.com/mxndtaylor/488db8ca053d8d69e72fa0fc4f5d0a23
